gateway/src/apicast/http_proxy.lua

```diff
@@ -132,8 +221,16 @@ local function forward_https_request(proxy_uri, proxy_auth, uri, skip_https_conn
     res, err = httpc:request(request)
 
     if res then
-        httpc:proxy_response(res)
-        httpc:set_keepalive()
+        if opts.request_unbuffered and is_chunked then
```

Comment:
Sorry, I do not get this. 

What is the relationship between request buffering option and response? Shouldn't the response be managed in the same way not matter existence of the unbuffered policy or being chunked request or not? i.e.

```
httpc:proxy_response(res)
httpc:set_keepalive()
```

Reply:
This is a bit tricky, after retrieve the raw socket, the above 2 methods will stop working, hence I need to send response back using the same raw socket

And we only use raw socket, when we have request_buffering option on

Reply:
the `proxy_response` method has been depreated. https://github.com/ledgetech/lua-resty-http#deprecated
So, we may need to implement this anyway. Maybe for the refactor to resty_http v0.17.0 PR.